feat(CoachmarkOverlayElements): Add next/back callbacks and currentStep properties

gcattan opened the pull request:
Closes https://github.com/carbon-design-system/ibm-products/issues/5596

This PR:
- Adds callbacks when clicking on Next and Previous buttons.
- Allows to set the current progression.

#### What did you change?

Add three optional properties to CoachmarkOverlayElements:
- onNextCallback
- onBackCallback
- currentStep

#### How did you test and verify your work?

I modified the story book as follows:

![image](https://github.com/user-attachments/assets/502ea220-26f5-42a3-969a-bd7184872adc)

Then manually checked that:
- next and back callbacks correctly logged into the console
- the progression was correctly set.

gcattan commented:
@AlexanderMelox CI looks good. But I still cannot access the logs of `percy`. Can you help with this issue?

AlexanderMelox commented:
> @AlexanderMelox CI looks good. But I still cannot access the logs of `percy`. Can you help with this issue?

I just got access to Percy so i can approve now

gcattan commented:
@AlexanderMelox  Following our suggestion and summarizing here ou exchange on slack: 
- I renamed the callbacks to `onNext` and `onBack`
- Added a test for the `onNext`.

However, testing the `onBack` is a bit more complicated as it is only showing when `scrollPercent > 0`, but this value never changes in testing (not sure why though).

Anyway, this helped me fix a bug with the scroll position not being correct (the position of the carrousel was not correctly updated).

AlexanderMelox commented:
> @AlexanderMelox Following our suggestion and summarizing here ou exchange on slack:
> 
> * I renamed the callbacks to `onNext` and `onBack`
> * Added a test for the `onNext`.
> 
> However, testing the `onBack` is a bit more complicated as it is only showing when `scrollPercent > 0`, but this value never changes in testing (not sure why though).
> 
> Anyway, this helped me fix a bug with the scroll position not being correct (the position of the carrousel was not correctly updated).

Jest doesn't really update css styles in testing so thats why it can be a little tricky to test scrolling and such, but we can handle it from now on
